trajectory_executor/odrive_test.py

```diff
@@ -0,0 +1,92 @@
+import rclpy
+from rclpy.node import Node
+
+from trajectory_interfaces.msg import JointAngles
+from trajectory_interfaces.srv import SensorDataRequest
+
+import serial
+import random
+import time
+
+
+class OdriveTest(Node):
+  """Tests sending a setpoint to the arduino/odrive and receiving tracking information"""
+  def __init__(self):
+    super().__init__('odrive_text')
+
+    self.publisher = self.create_publisher(
+      JointAngles,
+      'joint_angles',
+      10)
+
+    self.srv_client = self.create_client(
+      SensorDataRequest,
+      'sensor_data_request')
+
+    while not self.srv_client.wait_for_service(timeout_sec=1.0):
+      self.get_logger().info("Serial service inactive, waiting...")
+
+    self.req = SensorDataRequest.Request()
+
+  def send_request(self):
+    self.req.requested = True
+    self.future = self.srv_client.call_async(self.req)
+
+
+def main(args=None):
+  rclpy.init(args=args)
+  odrive_tester = OdriveTest()
+
+  angles = JointAngles()
+
+  while rclpy.ok():
+    angles.left_hip = random.randint(0, 359)
+    angles.left_knee = random.randint(0, 359)
```

Comment:
How are you verifying that the Physical motors actually move these angles? It would be hard to gauge it like this with random numbers right? Did you test it with the deterministic angle values?

Reply:
I did some testing with deterministic values and it worked fine. The verification that the motors get to where they're supposed to be comes from the arduino. The arduino checks the current position given by the ODrive against the goal position given by this node and sets an `at_goal` flag when the two values match up.